### PR TITLE
WINC-416: Improve logging and events

### PR DESCRIPTION
**Pull request by aravindhp**
- [wmco] Improve logging
- [wmco] Improve event messages

### CODE REVIEW

File: pkg/controller/windowsmachine/windows/windows.go [hunk 1]
```diff
@@ -487,7 +497,7 @@ func (vm *windows) getSourceVIP() (string, error) {
 		"where { $_.NetAdapter.LinkLayerAddress -eq $endpoint.MacAddress }).IPV4Address.IPAddress.Trim()\""
 	out, err := vm.Run(cmd, true)
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to get source VIP with output: %s", out)
+		return "", errors.Wrapf(err, "failed to get source VIP")
```
Comment (ravisantoshgudimetla) [inline pkg/controller/windowsmachine/windows/windows.go:500]: I think wrap might be sufficient?

File: pkg/controller/windowsmachine/windowsmachine_controller.go [hunk 2]
```diff
@@ -343,32 +344,32 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
...
 	// We always want the last entry which is the instanceID, and the first which is the provider name.
 	providerTokens := strings.Split(providerID, "/")
 	instanceID := providerTokens[len(providerTokens)-1]
 	if len(instanceID) == 0 {
-		return reconcile.Result{}, nil
+		return reconcile.Result{}, errors.Errorf("unable to get instance ID from provider ID for machine %s", machine.Name)
 	}
 	providerName := strings.TrimSuffix(providerTokens[0], ":")
 
+	log.Info("processing", "namespace", request.Namespace, "name", request.Name)
```
Comment (ravisantoshgudimetla) [inline pkg/controller/windowsmachine/windowsmachine_controller.go:358]: This can go in first.

**Review by ravisantoshgudimetla — COMMENTED** [review]
LGTM, except for some nits.

File: pkg/controller/windowsmachine/windows/windows.go [hunk 3]
```diff
@@ -487,7 +497,7 @@ func (vm *windows) getSourceVIP() (string, error) {
 		"where { $_.NetAdapter.LinkLayerAddress -eq $endpoint.MacAddress }).IPV4Address.IPAddress.Trim()\""
 	out, err := vm.Run(cmd, true)
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to get source VIP with output: %s", out)
```
[the remark this answers is not in the document]
Comment (aravindhp) [reply pkg/controller/windowsmachine/windows/windows.go:490]: As in `log.error`?

File: pkg/controller/windowsmachine/windowsmachine_controller.go (hunk 2 above)
Comment (aravindhp) [reply pkg/controller/windowsmachine/windowsmachine_controller.go:358]: I don't know what you mean by that. Do you want a separate PR for this?

File: pkg/controller/windowsmachine/windows/windows.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply pkg/controller/windowsmachine/windows/windows.go:490]: No we don't want to log anything here. Any errors during `vm.Run()` are being logged in the function itself.

File: pkg/controller/windowsmachine/windows/windows.go [hunk 4]
```diff
@@ -349,8 +357,7 @@ func (vm *windows) initializeBootstrapperFiles() error {
 	// Download the worker ignition to C:\Windows\Temp\ using the script that ignores the server cert
 	ignitionFileDownloadCmd := wgetIgnoreCertCmd + " -server " + vm.workerIgnitionEndpoint + " -output " +
 		winTemp + "worker.ign" + " -acceptHeader " + ignitionAcceptHeaderSpec
-	out, err := vm.Run(ignitionFileDownloadCmd, true)
-	log.V(1).Info("ignition file download", "cmd", ignitionFileDownloadCmd, "output", out)
```
Comment (mansikulkarni96) [inline pkg/controller/windowsmachine/windows/windows.go:353]: wouldn't having the out be helpful or we could log the error for vm.Run as well

File: pkg/controller/windowsmachine/windows/windows.go [hunk 5]
```diff
@@ -380,10 +387,11 @@ func (vm *windows) createService(svc *service) error {
 	if svc == nil {
 		return errors.New("service object should not be nil")
 	}
-	out, err := vm.Run("sc.exe create "+svc.name+" binPath=\""+svc.binaryPath+" "+
-		svc.args+" start=auto", false)
+	svcCreateCmd := "sc.exe create " + svc.name + " binPath=\"" + svc.binaryPath + " " + svc.args + " start=auto"
+	out, err := vm.Run(svcCreateCmd, false)
 	if err != nil {
-		return errors.Wrapf(err, "failed to create service with output: %s", out)
+		log.Info("failed to create service", "cmd", svcCreateCmd, out)
```
Comment (mansikulkarni96) [inline pkg/controller/windowsmachine/windows/windows.go:393]: odd number of log Info pairs.

**Review by mansikulkarni96 — COMMENTED** [review]
Thanks for working on this @aravindhp this is very helpful, PTAL at my comments.

File: pkg/controller/windowsmachine/windows/windows.go (hunk 4 above)
Comment (aravindhp) [reply pkg/controller/windowsmachine/windows/windows.go:353]: We are always [logging](https://github.com/openshift/windows-machine-config-operator/pull/147/files/9e894157568ff4faa08c5b95bf8291c38f4ec21b#diff-99dde3a6fcbd3bbbd7531252919d742735241faed60cf190a22c6c485ef92889R166) errors in `vm.Run`. Not sure if we need to do that here again.

File: pkg/controller/windowsmachine/windowsmachine_controller.go [hunk 6]
```diff
@@ -303,13 +309,15 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 					r.recorder.Eventf(machine, core.EventTypeWarning, "MachineDeletionFailed", "Machine %v deletion failed: unable to delete Machine object: %v", machine.Name, err)
 					return reconcile.Result{}, err
 				}
+				log.Info("machine has been remediated by deletion", "name", machine.GetName())
 				r.recorder.Eventf(machine, core.EventTypeNormal, "MachineDeleted", "Machine %v has been remediated by requesting to delete Machine object", machine.Name)
 				return reconcile.Result{}, nil
 			}
 			// version annotation exists with a valid value, node is fully configured, do nothing.
```
Comment (VaishnaviHire) [inline pkg/controller/windowsmachine/windowsmachine_controller.go:320]: Should we add a log message/ event here ?

**Review by VaishnaviHire — COMMENTED** [review]
Thanks @aravindhp for working on this.

LGTM except one comment.

File: pkg/controller/windowsmachine/windowsmachine_controller.go (hunk 6 above)
Comment (aravindhp) [reply pkg/controller/windowsmachine/windowsmachine_controller.go:320]: Sure I can add a log message here. Do you want it to be `log.Info()` or `log.V(1).Info()`? I don't think an event is warranted unless you can think of a reason a customer would want to see it.

File: pkg/controller/windowsmachine/windows/windows.go (hunk 5 above)
Comment (aravindhp) [reply pkg/controller/windowsmachine/windows/windows.go:393]: Looking at this log again, I should remove it. The error will be logged by `vm.Run()`.

File: pkg/controller/secret/secret_controller.go [hunk 7]
```diff
@@ -155,22 +155,22 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 	err = r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: userDataSecret, Namespace: userDataNamespace}, userData)
 	if err != nil && k8sapierrors.IsNotFound(err) {
 		// Secret is deleted
-		reqLogger.Info("UserData secret not found, creating the secret.")
+		log.Info("windows-user-data secret not found, creating the secret")
```
Comment (sebsoto) [inline pkg/controller/secret/secret_controller.go:158]: you can use `secrets.userDataSecret` here

Comment (sebsoto) [reply pkg/controller/secret/secret_controller.go:158]: applies to all uses in the file

File: pkg/controller/windowsmachine/nodeconfig/nodeconfig.go [hunk 8]
```diff
@@ -118,7 +125,7 @@ func getClusterAddr(kubeAPIServerEndpoint string) (string, error) {
 
 	// Check if hostname is valid
 	if !strings.HasPrefix(hostName, "api-int.") {
-		return "", errors.New("invalid API server url format found: expected hostname to start with `api-int.`")
+		return "", errors.New("invalid API server url format: expected hostname to start with `api-int.`")
```
Comment (sebsoto) [inline pkg/controller/windowsmachine/nodeconfig/nodeconfig.go:128]: should the actual hostname be added to this log message?
```
... invalid API server url '%s': expected hostname to start with `api-int.`", hostName)
```

File: pkg/controller/secret/secret_controller.go (hunk 7 above)
Comment (aravindhp) [reply pkg/controller/secret/secret_controller.go:158]: I thought about this but the log becomes a bit wordy and does not read well. Are you okay with `log.Info("secret not found, creating the secret", "name", userDataSecret)`?

Comment (sebsoto) [reply pkg/controller/secret/secret_controller.go:158]: Sounds good to me
